components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/part/DeleteElementAction.java

```diff
@@ -162,4 +194,41 @@ private boolean canDelete(EditPart editPart) {
         return true;
     }
 
+    protected <E extends EObject> boolean executeAddValueCommand(final EList<E> list, final E value,
+            boolean addToFront) {
+        TransactionalEditingDomain editingDomain = getDiagramEditor().getEditingDomain();
+        RecordingCommand command;
+        if (!addToFront) {
+            command = new RecordingCommand(editingDomain) {
+                protected void doExecute() {
+                    list.add(value);
+                }
+            };
+        } else {
+            command = new RecordingCommand(editingDomain) {
+                protected void doExecute() {
+                    list.add(0, value);
+                }
+            };
+        }
+        if (command.canExecute()) {
+            editingDomain.getCommandStack().execute(command);
+            return true;
+        }
+        return false;
+    }
+    
+    public static EsbDiagramEditor getDiagramEditor() {
```

Comment:
Since we use this method only within the class shall we make it private?

Reply:
Sure